src/main/java/TaskListener.java

```diff
@@ -0,0 +1,241 @@
+                    .substring("/from".length()).trim();
+
+            String toArgumentValue = arguments.substring(toCommandIndex)
+                    .substring("/to".length()).trim();
+
+            if (name.isEmpty()) {
+                printLines("Please provide the name argument");
+                return;
+            }
+
+            if (fromArgumentValue.isEmpty()) {
+                printLines("Please provide the /from argument");
+                return;
+            }
+
+            if (toArgumentValue.isEmpty()) {
+                printLines("Please provide the /to argument");
+                return;
+            }
+
+            Event event = new Event(name, false, fromArgumentValue, toArgumentValue);
+            this.tasksList.addTask(event);
+
+            printAddedTask(event);
+
+            tasksLoader.writeToFile(tasksList);
+        }
+    }
+
+    private void modifyDoneValue(boolean isDone, int taskIndex) {
+        String successMessage = isDone ? "Nice! I've marked this task as done:"
+                : "OK, I've marked this task as not done yet:";
+
+        try {
+            Task task = this.tasksList.getTasks().get(taskIndex);
+            task.setIsDone(isDone);
+
+            printLines(successMessage, task.toString());
+
+            tasksLoader.writeToFile(tasksList);
+        } catch (IndexOutOfBoundsException e) {
+            printLines("No task at such index!");
+        }
+    }
+
+    private void removeTask(int taskIndex) {
+        try {
+            Task removedTask = this.tasksList.removeTask(taskIndex);
+
+            String tasksWord = tasksList.getTasksCount() == 1 ? " task " : " tasks ";
+
+            String successMessage =  "Noted. I've removed this task:" +
+                    System.lineSeparator() +
+                    removedTask +
+                    System.lineSeparator() +
+                    "Now you have " + tasksList.getTasksCount() + tasksWord + "in the list.";
+
+           printLines(successMessage);
+        } catch (IndexOutOfBoundsException e) {
+            printLines("No task at such index!");
+        }
+    }
+    public void listen() {
+        Scanner in = new Scanner(System.in);
+
+        String line = in.nextLine();
+        String[] inputArray = line.split("\\s+", 2);
+
+        String command = inputArray[0];
+        String arguments = inputArray.length > 1 ? inputArray[1] : "";
+
+        switch (command) {
+
+        case "add":
+            addGenericTask(arguments);
+            break;
+
+        case "todo":
+            createTodo(arguments);
+            break;
+
+        case "deadline":
+            createDeadline(arguments);
+            break;
+
+        case "event":
+            createEvent(arguments);
+            break;
+
+        case "list": {
+            printLines(this.tasksList.toString());
+            break;
+        }
+
+        case "mark":
+        case "unmark": {
+            // isDone value that user wants to achieve
```

Comment:
Comments should not be used to explain a variable. You can remove this comment if it is not necessary